plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1988,14 +1988,19 @@ public Answer copyVolumeFromPrimaryToPrimary(CopyCommand cmd) {
             }
 
             destPool = storagePoolMgr.getStoragePool(destPrimaryStore.getPoolType(), destPrimaryStore.getUuid());
-            storagePoolMgr.copyPhysicalDisk(volume, destVolumeName, destPool, cmd.getWaitInMillSeconds());
-
-            if (srcPrimaryStore.isManaged()) {
-                storagePoolMgr.disconnectPhysicalDisk(srcPrimaryStore.getPoolType(), srcPrimaryStore.getUuid(), srcVolumePath);
-            }
+            try {
+                storagePoolMgr.copyPhysicalDisk(volume, destVolumeName, destPool, cmd.getWaitInMillSeconds());
+            } catch (Exception e) {
+                s_logger.debug("Failed to copy disk from src: " + volume.getPath() + " to dest: " +  destVolumeName);
+                throw new CloudRuntimeException(e.toString());
```

Comment:
why discard the old stack without logging it?
```suggestion
            } catch (Exception e) {
                s_logger.debug("Failed to copy disk from src: " + volume.getPath() + " to dest: " +  destVolumeName,e);
                throw new CloudRuntimeException(e.toString());
```
or
```suggestion
            } catch (Exception e) {
                s_logger.debug("Failed to copy disk from src: " + volume.getPath() + " to dest: " +  destVolumeName);
                throw new CloudRuntimeException(e);
```